lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteCacheFactory.java

```diff
@@ -18,33 +18,35 @@
 
 import com.futurewei.alcor.common.db.ICache;
 import com.futurewei.alcor.common.db.ICacheFactory;
-import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.Ignite;
+
 import javax.cache.expiry.CreatedExpiryPolicy;
 import javax.cache.expiry.Duration;
 import javax.cache.expiry.ExpiryPolicy;
 import java.util.concurrent.TimeUnit;
 
 public class IgniteCacheFactory implements ICacheFactory {
 
-    private IgniteClient igniteClient;
+    private Ignite ignite;
```

Comment:
@Gzure Could you think of renaming the class "ignite.Ignite" with more specific meaning, for example, IgniteXYZClient?

Reply:
Ignite.client.IgniteClient is a thin ignite client, about thin client [https://www.gridgain.com/docs/latest/developers-guide/thin-clients/getting-started-with-thin-clients](Thin Clients Overview). A  ignite. Ignite as an Ignite cluster node in client mode can support more functions than IgniteClient class, but it does not store data.

Reply:
Ignite.client.IgniteClient is a thin ignite client, about thin client [https://www.gridgain.com/docs/latest/developers-guide/thin-clients/getting-started-with-thin-clients](url). A ignite. Ignite as an Ignite cluster node in client mode can support more functions than IgniteClient class, but it does not store data.

Reply:
Ignite.client.IgniteClient is a thin ignite client, about thin client https://www.gridgain.com/docs/latest/developers-guide/thin-clients/getting-started-with-thin-clients. A ignite. Ignite as an Ignite cluster node in client mode can support more functions than IgniteClient class, but it does not store data.

Reply:
Okay. Thanks.

We might need to consider:
- Slow client handling (ref: https://apacheignite.readme.io/docs/clients-vs-servers#managing-slow-clients).
- Client reconnect handling (ref: https://apacheignite.readme.io/docs/clients-vs-servers#reconnecting-a-client). This requires adding a special exception handling in the catch block.
- Number of Ignite nodes in client mode. If the number goes large, it could affect the performance on the server side.

Reply:
ok